src/Libraries/Microsoft.Extensions.Telemetry.Testing/Metering/MetricCollector.cs

```diff
@@ -2,446 +2,336 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System;
-using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
 using System.Diagnostics.Metrics;
 using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
 using Microsoft.Shared.Diagnostics;
 
 namespace Microsoft.Extensions.Telemetry.Testing.Metering;
 
 /// <summary>
-/// The helper class to automatically capture metering data that has been recorded
-/// by instruments created by <see cref="Meter"/>.
+/// Collects the measurements published from an <see cref="Instrument{T}"/> or <see cref="ObservableInstrument{T}"/>.
 /// </summary>
-/// <remarks>
-/// This type has been designed to be used only for testing purposes.
-/// </remarks>
+/// <typeparam name="T">The type of metric data being recorded.</typeparam>
 [Experimental]
-[DebuggerDisplay("Count = {Count}")]
-public partial class MetricCollector : IDisposable
+[DebuggerDisplay("{_measurements.Count} measurements")]
+public sealed class MetricCollector<T> : IDisposable
```

Comment:
Now that this helper only deals in measurements rather than metrics, perhaps it should be called 'InstrumentRecorder' or 'MeasurementRecorder'. I suspect many devs won't notice the difference but we've tried to use the verb "Record" when saving or reporting raw measurements. For example it is `Instrument.Record()` and `MeterListener.RecordObservableInstruments()`. Whereas collection is described in [our metrics docs](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/metrics-collection): "The process of aggregating, transmitting, and storing data is called collection". `CollectedMeasurement` would be similarly updated to `RecordedMeasurement`.

Reply:
I'm going to keep collector as the moniker as it matches the terminology we use in other fakes, like the logging fakes and the redaction fakes.